packages/ui/src/domainModules/localityService/selectors.js

```diff
@@ -51,8 +51,23 @@ const createDropdownSelector = groupFilter => {
           }
         })
         .filter(({ text }) => {
-          return text.toLowerCase().includes(lowerCaseSearchQuery)
+          return text.toLowerCase().indexOf(lowerCaseSearchQuery) === 0
         })
+
+      const otherMatches = Object.values(curatedLocalities)
+        .filter(({ group }) => group === groupFilter)
+        .map(({ id, name }) => {
```

Comment:
Would make sense to save the result of this `map` in a variable and then apply the two filters on that, so we don't do the same Object.values-filter-map twice